modules/service/src/main/scala/lucuma/odb/graphql/mapping/ObservationMapping.scala

```diff
@@ -132,30 +134,61 @@ trait ObservationMapping[F[_]]
     effectHandler(readEnv, calculate)
   }
 
-  lazy val configurationRequestsQueryHandler: EffectHandler[F] = {
-    val readEnv: Env => Result[Unit] = _ => ().success
-
-    val calculate: (Program.Id, Observation.Id, Unit) => F[Result[List[ConfigurationRequest]]] =
-      (_, oid, _) =>
-        services.useTransactionally {
-          configurationService
-            .selectRequests(oid)
-        }
+  lazy val configurationRequestsQueryHandler: EffectHandler[F] = { pairs =>
```

Comment:
I don't think I'd seen this method of creating an EffectHandler before.

Reply:
`EffectHandler` is a single abstract method (SAM) interface so you can define it using function syntax. Scala has supported this for a long time but it's not commonly used.

Reply:
I only recently figured out that's also what's going on with `Matcher` implementations...